core/src/main/scala/org/bitcoins/core/wallet/utxo/TxoState.scala

```diff
@@ -0,0 +1,36 @@
+package org.bitcoins.core.wallet.utxo
+
+/** Represents the various states a transaction output can be in */
+sealed abstract class TxoState
+
+sealed abstract class ReceivedState extends TxoState
+
+sealed abstract class SpentState extends TxoState
+
+object TxoState {
+
+  /** Means that no funds have been sent to this utxo EVER */
+  final case object DoesNotExist extends TxoState
+
+  /** Means we have received funds to this utxo, but they are not confirmed */
+  final case object UnconfirmedReceived extends ReceivedState
+
+  /** Means we have received funds and they are fully confirmed for this utxo */
+  final case object ConfirmedReceived extends ReceivedState
+
+  /** Means we have spent this utxo, but it is not fully confirmed */
+  final case object UnconfirmedSpent extends SpentState
+
+  /** Means we have spent this utxo, and it is fully confirmed */
+  final case object ConfirmedSpent extends SpentState
+
```

Comment:
An immature state might be useful as well for txs that have been mined but not able to be spent yet

Reply:
Hmm... I guess that is what I meant by "Unconfirmed" but i see how this can be confusing if you take it literally. There are two "Unconfirmeds" -- unconfirmed in the network and unconfirmed in your wallet. The latter setting is dependent on how many confirmations you think your transaction needs before it is fully "settled".